pytests/test_socket.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root
+# for license information.
+
+import pytest
+from ptvsd.socket import create_server, shut_down
+
+
+class TestSocketServerReuse(object):
+    HOST1 = '127.0.0.1'
+    HOST2 = '127.0.0.2'
+    PORT1 = 7890
+
+    def test_reuse_same_address_port(self):
+        sock1 = create_server(self.HOST1, self.PORT1)
+        with pytest.raises(Exception):
+            create_server(self.HOST1, self.PORT1)
+        assert sock1.getsockname() == (self.HOST1, self.PORT1)
+        shut_down(sock1)
```

Comment:
You might want to do these in `finally`, so that they get closed even if the test fails.